### PR TITLE
Add codeql, scorecard, and verifying license

### PR DESCRIPTION
**Pull request by Hayden-IO**

### CODE REVIEW

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+jobs:
+  analysis:
+    name: Scorecards analysis
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+    uses: sigstore/community/.github/workflows/reusable-scorecard.yml@d0c95c8803672313d0bf72e1a44021be5b583c24 # main
```
Comment (bobcallaway) [inline .github/workflows/scorecard.yml]: i'd just pin to main since we're managing this within the community.

Comment (Hayden-IO) [reply .github/workflows/scorecard.yml]: Done

**Review by phillmv — APPROVED** [review]
lgtm!

**Review by codysoyland — APPROVED** [review]

**Comment by codysoyland** [comment]
The linter is only scheduled to run [when go code changes](https://github.com/sigstore/sigstore-go/blob/main/.github/workflows/golangci-lint.yml#L21-L24), but it's a required check. Do you want to just remove that constraint so it runs every time? CodeQL will always take longer anyway.

**Comment by Hayden-IO** [comment]
Ah good catch! I thought it was an issue with the import. I’ll get that removed.

**Review by codysoyland — APPROVED** [review]

**Pull request merged**